### ISSUE TITLE
Suggestion: Debuggers

### ISSUE THREAD

**Issue by AngeloG** [Overv/Open.GL#54]
The chapter about framebuffers suggests this:

>Try changing the clear color of the scene and reading it back using glReadPixels to check if the scene renders properly to the new framebuffer.

Why not add a note somewhere about graphics debuggers? For example, I've been learning a lot from just looking at the output of [CodeXL](https://github.com/GPUOpen-Tools/CodeXL):

![](https://4o4.nl/20160430ZvCJZG61.png)

I don't have to worry about writing extra code with `glReadPixels`, I can just look at my debugger to make sure it's rendered okay. I guess using `glReadPixels` is a good excercise in itself, though.

**Comment by Overv** [comment]
Yeah, I really regret not writing about `glDebugMessageCallback​` and using CodeXL/RenderDoc, but I feel that it is a little bit too late to integrate that now.
